core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -86,6 +89,15 @@ public CallOptions withAuthority(@Nullable String authority) {
   }
 
   /**
+   * Returns a new {@code CallOptions} with the given call credentials.
+   */
+  public CallOptions withCallCredentials(@Nullable CallCredentials credentials) {
```

Comment:
`@ExperimentalApi` since `CallCredentials` is experimental. Ditto for the stub. Ditto for `getCallCredentials()`.